### PR TITLE
unit tests for monitoring schedule

### PR DESCRIPTION
**Pull request by ryansteakley**
Description of changes:

- sdk.go coverage 70.3%
- Custom code hooks 100%
- Overall custom code > 95%

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/resource/monitoring_schedule/testdata/sdkapi/update/update_success.json [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            }
+        },
+        "MonitoringJobDefinitionName": null,
+        "MonitoringType": "DataQuality",
+        "ScheduleConfig": {
+            "ScheduleExpression": "cron(0 * ? * * *)"
+        }
+    },
+    "MonitoringScheduleName": "unit-testing-monitoring-schedule",
+    "MonitoringScheduleStatus": "Pending",
```
Comment (mbaijal) [inline pkg/resource/monitoring_schedule/testdata/sdkapi/update/update_success.json:76]: same filename comment, we should ideally not call the Pending status as sucess.

File: pkg/resource/monitoring_schedule/testdata/test_suite.yaml [hunk 2]
```diff
@@ -1,17 +1,168 @@
...
+          - operation: DescribeMonitoringScheduleWithContext
+            output_fixture: "sdkapi/describe/success_describe.json"
+      invoke: ReadOne
+      expect:
+        latest_state: "v1alpha1/readone/observed/conditions_clear_on_success.yaml"
+  - name: "Monitoring Schedule update tests"
+    description: "Testing the update operation"
+    scenarios:
+    - name: "Update=RequeueOnCreating"
+      description: "This test checks if the controller requeues if monitoring schedule is in pending status and Job is InProgress"
```
Comment (mbaijal) [inline pkg/resource/monitoring_schedule/testdata/test_suite.yaml:88]: how exactly are we checking that the controller requeued here ?

File: pkg/resource/monitoring_schedule/testdata/test_suite.yaml [hunk 3]
```diff
@@ -1,17 +1,168 @@
...
+              message: "Server is down"
+      invoke: Update
+      expect:
+        latest_state: "v1alpha1/update/observed/error_on_update.yaml"
+        error: "ServiceUnavailable: Server is down\n\tstatus code: 0, request id: "
+  - name: "Monitoring Schedule delete tests"
+    description: "Testing the delete operation"
+    scenarios:
+    - name: "Delete=RequeueOnCreating"
+      description: "This test checks if the controller requeues if monitoring schedule is in pending status and Job is InProgress"
```
Comment (mbaijal) [inline pkg/resource/monitoring_schedule/testdata/test_suite.yaml:124]: same question - how are we checking controller requeue - the description seems inaccurate ?

File: pkg/resource/monitoring_schedule/testdata/test_suite.yaml [hunk 4]
```diff
@@ -1,17 +1,168 @@
...
+          - operation: CreateMonitoringScheduleWithContext
+            output_fixture: "sdkapi/create/success_create.json"
+      invoke: Create
+      expect:
+        latest_state: "v1alpha1/create/observed/success_after_create.yaml"
+        error: null
+  - name: "Monitoring Schedule readOne tests"
+    description: "Testing the readOne operation"
+    scenarios:
+    - name: "ReadOne=MissingRequiredField"
```
Comment (akartsky) [inline pkg/resource/monitoring_schedule/testdata/test_suite.yaml:32]: nit : Indent each of these scenarios

File: pkg/resource/monitoring_schedule/testdata/test_suite.yaml [hunk 5]
```diff
@@ -1,17 +1,168 @@
...
+      invoke: ReadOne
+      expect:
+        latest_state: "v1alpha1/readone/observed/scheduled.yaml"
+    - name: "ReadOne=SuccessClearsConditions"
+      description: Testing a successful reconciliation clears conditions if terminal/recoverable condition were already set to true
+      given:
+        desired_state: "v1alpha1/readone/desired/error_conditions_true.yaml"
+        svc_api:
+          - operation: DescribeMonitoringScheduleWithContext
+            output_fixture: "sdkapi/describe/success_describe.json"
```
Comment (akartsky) [inline pkg/resource/monitoring_schedule/testdata/test_suite.yaml:80]: why does it reconcile if terminal state is true ? 
(same question for other PRs)

File: pkg/resource/monitoring_schedule/testdata/test_suite.yaml (hunk 2 above)
Comment (ryansteakley) [reply pkg/resource/monitoring_schedule/testdata/test_suite.yaml:88]: checking by seeing that the requeueOnCreating err is thrown

**Pull request merged**